src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/DesignTimeInputsChangeTracker.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    internal class DesignTimeInputsChangeTracker : ProjectValueDataSourceBase<DesignTimeInputsDelta>, IDesignTimeInputsChangeTracker
+    {
+        private readonly UnconfiguredProject _project;
+        private readonly IProjectThreadingService _threadingService;
+        private readonly IActiveConfiguredProjectSubscriptionService _projectSubscriptionService;
+        private readonly IDesignTimeInputsDataSource _inputsDataSource;
+        private readonly IDesignTimeInputsFileWatcher _fileWatcher;
+
+        private readonly DisposableBag _disposables = new DisposableBag();
+
+        private ITargetBlock<IProjectVersionedValue<Tuple<DesignTimeInputs, IProjectSubscriptionUpdate>>>? _inputsActionBlock;
+        private ITargetBlock<IProjectVersionedValue<string[]>>? _fileWatcherActionBlock;
+
+        private DesignTimeInputs? _latestDesignTimeInputs;
+        private string? _latestOutputPath;
+        private int _version;
+
+        /// <summary>
+        /// The block that sends out updates from this service.
+        /// </summary>
+        private IBroadcastBlock<IProjectVersionedValue<DesignTimeInputsDelta>>? _broadcastBlock;
+
+        /// <summary>
```

Comment:
I don't find these comments to be helpful - they are just restating their names.

Reply:
Thanks, removed one, expanded another.